src/Microsoft.AspNetCore.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -29,18 +29,17 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            // Ignore the passed in content type, if the object is string
-            // always return it as a text/plain format.
             if (context.ObjectType == typeof(string) || context.Object is string)
             {
-                if (!context.ContentType.HasValue)
+                // Call into base to check if the current request's content type is a supported media type.
+                var canWriteResult = base.CanWriteResult(context);
+                if (canWriteResult)
                 {
                     var mediaType = SupportedMediaTypes[0];
                     var encoding = SupportedEncodings[0];
                     context.ContentType = new StringSegment(MediaType.ReplaceEncoding(mediaType, encoding));
+                    return true;
```

Comment:
It would be nice to see in comments here a truth table about what the parameters are for this decision and what the expected output is.

My understanding is that this formatter will only return true when both of the following are true:
 - The data type is string
 - The content type is `null` **OR** the content type is in `SupportedMediaTypes`. 

So what happens if the user adds `application/json` to this formatter because they **want** to return preformatted json blobs from actions? This code would turn it into text/plain.